cogs/challenges.py

```diff
@@ -54,17 +52,24 @@ async def on_message(self, message):  # Submitted role.
 
             if submitted not in message.author.roles:
                 await message.delete()
-                if message.content.count("```") != 2:
-                    msg = f"{message.author.mention} make sure to submit in a code " \
-                          f"block and only include the code required for the challenge!" \
-                          f"\nUse `tim.tag discord code` for more information!"
+                attach = message.attachments and message.attachments[0]
+                
+                if not attach:
+                    msg = f"{message.author.mention} make sure to __upload a " \
+                          f"python file__ that only includes the code required " \
+                          f"for the challenge!"
+                    return await message.channel.send(msg, delete_after=10.0)
+                
+                content = "```py\n" + (await attach.read()).decode("u8").replace("`", "\u200b`") + "```"
+                if len(content) > 2040:
+                    msg = f"{message.author.mention} attachment can't be __more" \
+                          f" than 2040 characters__."
                     return await message.channel.send(msg, delete_after=10.0)
 
                 await message.author.add_roles(submitted)
-                embed = discord.Embed(description=message.content,
-                                      color=message.guild.me.top_role.color)
```

Comment:
change the color to something like `message.author.top_role.color` instead of removing it :stonks:

Reply:
Nah i prefer all embeds having the same color.
Maybe i use `0x36393E`

Reply:
> Nah i prefer all embeds having the same color.
> Maybe i use `0x36393E`

Maybe if this was ur bot 👀
Stick to the usage of the project

Reply:
> Maybe if this was ur bot 👀
Stick to the usage of the project

It was `message.guild.me.top_role.color` and the bot's top role color was the default one `#000000`. Thats why all the embeds in the hidden channel are black :mhm:
So do you want to keep it black like that or change it to `message.guild.me.color` (Pink `#FF1955`, the currently it have rn) or change it to the same color we using for help command (`#36393E`) or change it to `message.author.color` or ...

Reply:
I’d say `message.author.color` but invisible is good too (`#36393E`)